proposed/2023/nuget-package-search.md

```diff
@@ -0,0 +1,72 @@
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+#### 1. Adding `-ExactMatch` Option: 
+A new flag, `-ExactMatch`, will be added to the `search` command. When this flag is used, it will only return packages whose names exactly match the query.
+- **Version not specified** : In this case, the latest version of the package specified is listed .\
+*Nuget.exe Usage* : `nuget search <PackageName> -ExactMatch` \
+*Dotnet Usage* : `dotnet nuget search <PackageName> -ExactMatch`
+- **Version specified** : In this case, the package with the specific version number is listed.\
+*Nuget.exe Usage* : `nuget search <PackageName> -ExactMatch -Version 1.2.3.4` \
+*Dotnet Usage* : `dotnet nuget search <PackageName> -ExactMatch -Version 1.2.3.4`
+- **All Versions** : In this case, all versions of the package with the exact match are listed.\
+*Nuget.exe Usage* : `nuget search <PackageName> -ExactMatch -AllVersions` \
+*Dotnet Usage* : `dotnet nuget search <PackageName> -ExactMatch -AllVersions`
+#### 2. Deprecating list Command: 
+The list command is planned to be deprecated, given its functionalities can be covered by the search command. A warning message will be displayed to inform users of this change. `Warning: The 'list' command is deprecated. Use 'search' instead.` \
+The functionalities of the `list` command can be replicated as follows
+* In nuget.exe `nuget list` will be `nuget search`.
+* In Dotnet, `dotnet nuget search` will be added and it will be the same as `nuget search`.
+#### 3. Introducing nuget search to Dotnet:
+A new command dotnet nuget search will be introduced to bring package search functionality to the Dotnet CLI. This will help in resolving the highly requested `dotnet nuget list` functionality.
```

Comment:
```suggestion
A new command `dotnet nuget search` will be introduced to bring package search functionality to the Dotnet CLI. This will help in resolving the highly requested `dotnet nuget list` functionality.
```